pandas/tests/test_groupby.py

```diff
@@ -5053,6 +5053,15 @@ def test_groupby_categorical_two_columns(self):
                          "C3":[nan,nan,nan,nan, 10,100,nan,nan, nan,nan,200,34]}, index=idx)
         tm.assert_frame_equal(res, exp)
 
+    def test_groupby_apply_all_none(self):
+        # Tests to make sure no errors if apply function returns all None 
+        # values. Issue 9684.
+        test_df = pd.DataFrame({'groups': [0,0,1,1], 'random_vars': [8,7,4,5]})
+        
+        def test_func(x):
+            pass
+        test_df.groupby('groups').apply(test_func)
```

Comment:
compare the returned results to an expectation here. pls also squash and rebase on master.